pkg/btf/btf_test.go

```diff
@@ -178,34 +178,45 @@ func genericTestFindBtfFuncParamFromHook(t *testing.T, hook string, argIndex int
 	return nil
 }
 
-func TestFindBtfFuncParamFromHook(t *testing.T) {
-	// Assert no errors on Kprobe
-	hook := "wake_up_new_task"
-	argIndex := 0
-	expectedName := "p"
-	err := genericTestFindBtfFuncParamFromHook(t, hook, argIndex, expectedName)
-	assert.NoError(t, err)
+func testFindBtfFuncParamFromHook(btfFName string) func(*testing.T) {
+	return func(t *testing.T) {
+		spec, err := btf.LoadSpec(btfFName)
+		if err != nil {
+			t.Skipf("%q not found", btfFName)
+		}
+
+		// Assert no errors on Kprobe
+		hook := "wake_up_new_task"
+		argIndex := 0
+		expectedName := "p"
+		err = genericTestFindBtfFuncParamFromHook(t, spec, hook, argIndex, expectedName)
+		assert.NoError(t, err)
 
-	// Assert error raises with invalid hook
-	hook = "fake_hook"
-	argIndex = 0
-	expectedName = "p"
-	err = genericTestFindBtfFuncParamFromHook(t, hook, argIndex, expectedName)
-	assert.ErrorContains(t, err, fmt.Sprintf("failed to find BTF type for hook %q", hook))
-
-	// Assert error raises when hook is a valid BTF type but not btf.Func
-	hook = "linux_binprm"
-	argIndex = 0
-	expectedName = "p"
-	err = genericTestFindBtfFuncParamFromHook(t, hook, argIndex, expectedName)
-	assert.ErrorContains(t, err, fmt.Sprintf("failed to find BTF type for hook %q", hook))
-
-	// Assert error raises when argIndex is out of scope
-	hook = "wake_up_new_task"
-	argIndex = 10
-	expectedName = "p"
-	err = genericTestFindBtfFuncParamFromHook(t, hook, argIndex, expectedName)
-	assert.ErrorContains(t, err, fmt.Sprintf("index %d is out of range", argIndex))
+		// Assert error raises with invalid hook
+		hook = "fake_hook"
+		argIndex = 0
+		expectedName = "p"
+		err = genericTestFindBtfFuncParamFromHook(t, spec, hook, argIndex, expectedName)
+		assert.ErrorContains(t, err, fmt.Sprintf("failed to find BTF type for hook %q", hook))
+
+		// Assert error raises when hook is a valid BTF type but not btf.Func
+		hook = "linux_binprm"
+		argIndex = 0
+		expectedName = "p"
+		err = genericTestFindBtfFuncParamFromHook(t, spec, hook, argIndex, expectedName)
+		assert.ErrorContains(t, err, fmt.Sprintf("failed to find BTF type for hook %q", hook))
+
+		// Assert error raises when argIndex is out of scope
+		hook = "wake_up_new_task"
+		argIndex = 10
+		expectedName = "p"
+		err = genericTestFindBtfFuncParamFromHook(t, spec, hook, argIndex, expectedName)
+		assert.ErrorContains(t, err, fmt.Sprintf("index %d is out of range", argIndex))
+	}
+}
+
+func TestFindBtfFuncParamFromHook(t *testing.T) {
+	t.Run(defaults.DefaultBTFFile, testFindBtfFuncParamFromHook(defaults.DefaultBTFFile))
```

Comment:
so we will eventually add more tests in here for other btf paths? could it be part of this change?

Reply:
Sure, I thought I would simply improve this behavior and in another PR upload additionnal BTF files. 
Though, I've added a function `listBtfFiles` that return the list of btf files to test. 

At first, I was thinking of crawling the directory where the below btf files are uploading to make an automatic check. But with this we have less granularity.

https://github.com/cilium/tetragon/blob/a10ac161e2292b54fda6030cb0da582a117888ce/Makefile#L307-L308

The btf files I've added were from `observerFindBTF` and the `TestSpecs`. I believe you were thinking of those files?

Reply:
> Sure, I thought I would simply improve this behavior and in another PR upload additionnal BTF files.
> Though, I've added a function listBtfFiles that return the list of btf files to test.

I'd suggest you already have the btf files? let's upload them to https://github.com/cilium/tetragon-testdata and put all pieces together

Reply:
I do not since this was not part of the initial request (from https://github.com/cilium/tetragon/pull/3143#discussion_r1948539484). I'll be happy to add any files that you could suggest. 

But what are the needs of having more btf files than those in `listBtfFiles` ?